themes/default/content/blog/pulumi-kubernetes-operator-1-0/index.md

```diff
@@ -0,0 +1,111 @@
+
+Since then, we’ve moved the Pulumi Kubernetes Operator over to use the Pulumi Automation API, significantly reducing the size of the operator implementation, building on a robust foundation shared with dozens of other projects in the Pulumi ecosystem, and benefiting from many improvements and features added to the Pulumi Automation API. Earlier this year, the [Pulumi Automation API]({{< relref "/blog/pulumiup-automation-api-ga" >}}) also reached GA as part of the [Pulumi 3.0]({{< relref "/blog/pulumi-3-0" >}}) release.
+
+### Support for any state and secrets management backend
+
+Pulumi supports a wide variety of state management and secrets management backends. Since the initial release of the Pulumi kubernetes Operator, we have added support for using not just the Pulumi Service backends, but also all of the other open source backend options, such as S3, Azure Blob and GCS for state storage, and KMS, KeyVault and Vault for secrets management.
+
+```yaml
+apiVersion: pulumi.com/v1
+kind: Stack
+metadata:
+  name: s3-bucket-stack
+spec:
+  backend: s3://my-s3-backend
+  stack: acmecorp/operator-dev
+  projectRepo: https://github.com/acmecorp/pytest1
+  initOnCreate: true
+  branch: refs/head/master
+  config:
+    aws:region: us-west-2
+```
+
+### Worry-free Upgrades
+
+The Pulumi Kubernetes Operator manages cloud infrastructure deployments which can take seconds to tens of minutes to complete. Although Pulumi ensures these deployments can be continued if interrupted, we want to ensure as little disruption to critical cloud infrastructure deployments as possible via the Operator. As a result, we have added a few key features to ensure that the Pulumi Kubernetes Operator can be safely upgraded to new versions without disrupting any deployments that are in-flight.
+
+This includes capabilities like [Leader Election](https://sdk.operatorframework.io/docs/building-operators/golang/advanced-topics/#leader-with-lease), [Graceful Shutdown](https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/manager#Options) and more.
+
+### Prometheus Metrics
+
+The Pulumi Kubernetes Operator automatically manages a lot of details of Pulumi deployments, and as a result, it’s critical to be able to get insight into the operation of the Operator. Previously, this was available either via operator logs, or via the Pulumi Console. As part of the 1.0 release, the Operator is also now exporting Prometheus-compatible metrics for `stacks_active` and `stacks_failing`. These can be used to drive reporting and other workflows inside the Kubernetes cluster.
+
+<!-- TODO: Screenshot or curl output of metrics? -->
+
+<!-- TODO: ### Documentation -->
```

Comment:
```suggestion
The Pulumi Kubernetes Operator automatically manages a lot of details of Pulumi deployments, and as a result, it’s critical to be able to get insight into the operation of the Operator. Previously, this was available either via operator logs, or via the Pulumi Console. As part of the 1.0 release, the Operator is also now exporting [Prometheus-compatible metrics](https://github.com/pulumi/pulumi-kubernetes-operator/pull/217#issue-1021521569) for `stacks_active` and `stacks_failing`. These can be used to drive reporting and other workflows inside the Kubernetes cluster.

<!-- TODO: ### Add metrics Documentation once https://github.com/pulumi/pulumi-kubernetes-operator/issues/218 is addressed. -->
```

There seem to be some issues with the service monitors configured by default by the operator framework for scraping metrics. I am tracking better out-of-the-box prometheus integration in https://github.com/pulumi/pulumi-kubernetes-operator/issues/218. Will follow-up with links separately when I fix that.